### PR TITLE
fix getting query segment info error during the period of loading and…

### PR DESCRIPTION
**Pull request by MrPresent-Han**
… unloading segments

issue: #20281 

Signed-off-by: MrPresent-Han <chun.han@zilliz.com>

### CODE REVIEW

File: internal/querycoordv2/handlers.go [hunk 1]
```diff
@@ -62,8 +62,20 @@ func (s *Server) checkAnyReplicaAvailable(collectionID int64) bool {
...
 	for _, segment := range segments {
+		if _, existCurrentTarget := currentTargetSegmentsMap[segment.GetID()]; !existCurrentTarget {
+			//if one segment exists in distMap but doesn't exist in currentTargetMap
+			//in order to guarantee that get segment request launched by sdk could get
+			//consistent result, for example
+			//sdk insert three segments:A, B, D, then A + B----compact--> C
+			//In this scenario, we promise that clients see either 2 segments(C,D) or 3 segments(A, B, D)
+			//rather than 4 segments(A, B, C, D), in which query nodes are loading C but have completed loading process
+			log.Info("filtered segment being in the intermediate status",
+				zap.Int64("segId", segment.GetID()))
```
Comment (yah01) [inline internal/querycoordv2/handlers.go:76]: log naming convention: camelCase, but full UPPERCASE for `ID`, so `segId` -> `segmentID`

Comment (MrPresent-Han) [reply internal/querycoordv2/handlers.go:76]: OK

**Comment by MrPresent-Han** [comment]
rerun ut

**Comment by MrPresent-Han** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**